### PR TITLE
Add tests for BaseModel

### PR DESCRIPTION
**Pull request by bdunne**

### CODE REVIEW

**Comment by bdunne** [comment]
@bzwei Please review
cc @syncrou

File: spec/base_model_spec.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    context "with json" do
+      let(:json_or_hash) { hash.to_json }
+      include_examples "validate new object"
+    end
+
+    it "doesn't define related attributes" do
+      hash_with_relations = hash.merge("host" => 1, "related" => {"host" => "http://example.com/hosts/1"})
+
+      expect(described_class.new(api, hash_with_relations)).to_not respond_to(:host)
```
Comment (bzwei) [inline spec/base_model_spec.rb]: should validate method `#host_id` and value.

File: spec/base_model_spec.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    {
+      "name"            => "jeff",
+      "address"         => { "street" => "22 charlotte rd"},
+      "serialized_json" => { "key" => "value"}.to_json,
+      "serialized_yaml" => { "key" => "value"}.to_yaml
+    }
+  end
+
+  context ".initialize" do
+    shared_examples_for "validate new object" do
```
Comment (syncrou) [inline spec/base_model_spec.rb]: I like how this is laid out. Easy to read in the context of the other tests in this file.

**Comment by syncrou** [comment]
👍 LGTM

**Pull request merged**